File: leetcode/easy/682/682.ts
function calPoints(operations: string[]): number {
  const stack: number[] = [];

  for (const op of operations) {
    if (op === "C") {
      stack.pop();
    } else if (op === "D") {
      const lastScore = stack[stack.length - 1];
      stack.push(lastScore * 2);
    } else if (op === "+") {
      const lastScore = stack[stack.length - 1];
      const secondLastScore = stack[stack.length - 2];
      stack.push(lastScore + secondLastScore);
    } else {
      stack.push(parseInt(op));
    }
  }

  let sum = 0;
  for (const score of stack) {
    sum += score;
  }
  return sum;
}

File: leetcode/easy/682/682.go
package main

import "strconv"

func calPoints(operations []string) int {
	stack := []int{}

	for _, op := range operations {
		switch op {
		case "C":
			stack = stack[:len(stack)-1]
		
		case "D":
			lastScore := stack[len(stack)-1]
			stack = append(stack, lastScore*2)
		
		case "+":
			lastScore := stack[len(stack)-1]
			secondLastScore := stack[len(stack)-2]
			stack = append(stack, lastScore+secondLastScore)
		
		default:
			num, _ := strconv.Atoi(op)
			stack = append(stack, num)
		}
	}

	sum := 0
	for _, score := range stack {
		sum += score
	}

	return sum
}